pkg/api/v1/get_routes.go

```diff
@@ -0,0 +1,232 @@
+package v1
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	"github.com/els0r/goProbe/pkg/capture"
+	"github.com/els0r/status"
+	"github.com/go-chi/chi"
+)
+
+func (a *API) getRequestRoutes(r chi.Router) {
+
+	// for statistics about the capture
+	r.Route("/stats", func(r chi.Router) {
+		r.Get("/packets", a.getPacketStats)
+		r.Get("/errors", a.getErrors)
+	})
+
+	// inspection of flows
+	r.Route("/flows", func(r chi.Router) {
+		r.Route("/{ifaceName}", func(r chi.Router) {
+			r.With(a.IfaceCtx).Get("/", a.getActiveFlows)
+		})
+	})
+}
+
+// The key type is unexported to prevent collisions with context keys defined in
+// other packages.
+type key int
+
+// context keys for this api
+const (
+	activeFlowsCtxKey key = iota
+)
+
+// IfaceCtx obtains the flow map for the queried interface and stores it in the request context
+func (a *API) IfaceCtx(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ifaceName := chi.URLParam(r, "ifaceName")
+
+		flows, err := a.c.ActiveFlows(ifaceName)
+		if err != nil {
+			http.Error(w, http.StatusText(404), 404)
+			return
+		}
+		ctx := context.WithValue(r.Context(), activeFlowsCtxKey, flows)
+		next.ServeHTTP(w, r.WithContext(ctx))
+	})
+}
+
+func (a *API) getActiveFlows(w http.ResponseWriter, r *http.Request) {
+	var err error
+
+	ctx := r.Context()
+	flowLog, ok := ctx.Value(activeFlowsCtxKey).(map[string]*capture.FlowLog)
+	if !ok {
+		http.Error(w, http.StatusText(422), 422)
+		return
+	}
+
+	// encode answer
+	if !printPretty(r) {
+		err := json.NewEncoder(w).Encode(&flowLog)
```

Comment:
As mentioned during our phone call, I'd replace all occurrences of encoding/json with jsoniter (https://github.com/json-iterator/go) for performance reasons.